frontend/pages/beta/schemas/new.tsx

```diff
@@ -0,0 +1,166 @@
+                </Typography>
+                <TextField
+                  fullWidth
+                  required
+                  size='small'
+                  value={schemaId}
+                  aria-label='Schema ID'
+                  onChange={(e) => setSchemaId(e.target.value)}
+                />
+                <Typography variant='caption'>Please specify a unique ID for your schema</Typography>
+              </Stack>
+              <Stack>
+                <Typography fontWeight='bold'>
+                  Name <span style={{ color: 'red' }}>*</span>
+                </Typography>
+                <TextField
+                  fullWidth
+                  required
+                  size='small'
+                  value={schemaName}
+                  aria-label='Schema name'
+                  onChange={(e) => setSchemaName(e.target.value)}
+                />
+                <Typography variant='caption'>Please specify a name for your schema</Typography>
+              </Stack>
+              <Stack>
+                <Typography fontWeight='bold'>
+                  Description <span style={{ color: 'red' }}>*</span>
+                </Typography>
+                <RichTextEditor
+                  value={schemaDescription}
+                  onChange={(input) => setSchemaDescription(input)}
+                  aria-label='Schema description'
+                />
+                <Typography variant='caption'>A short description describing the purpose of this schema</Typography>
+              </Stack>
+              <Stack>
+                <Typography fontWeight='bold'>
+                  Schema Type <span style={{ color: 'red' }}>*</span>
+                </Typography>
+                <TextField
+                  select
+                  size='small'
+                  required
+                  value={schemaKind}
+                  onChange={(event): void => setSchemaKind(event.target.value as SchemaKind)}
+                >
+                  <MenuItem value={SchemaKind.MODEL}>Model</MenuItem>
+                  <MenuItem value={SchemaKind.ACCESS}>Access Request</MenuItem>
+                </TextField>
+                <Typography variant='caption'>
+                  Schemas are used for both model cards and access request forms
+                </Typography>
+              </Stack>
+              <Button variant='outlined' component='label' aria-label='Schema JSON file upload button'>
+                {filename !== '' ? filename : 'Select schema'}
+                <input type='file' hidden onChange={handleUploadChange} />
+              </Button>
+              <Stack alignItems='flex-end'>
+                <LoadingButton
+                  variant='contained'
+                  loading={loading}
+                  type='submit'
+                  disabled={!schemaId || !schemaName || !schemaDescription || !jsonSchema}
+                  sx={{ width: 'fit-content' }}
```

Comment:
Is this needed? It should, by default, set its own width based on the size of the text